internal/utils/main.go

```diff
@@ -538,7 +539,15 @@ var futureArchiveURLs = []string{
 }
 
 func CreateHistoryArchiveClient(archiveURLS []string) (historyarchive.ArchiveInterface, error) {
-	return historyarchive.NewArchivePool(archiveURLS, historyarchive.ConnectOptions{})
+	var archiveInterface historyarchive.ArchiveInterface
+
+	archivePool, err := historyarchive.NewArchivePool(archiveURLS, historyarchive.ConnectOptions{UserAgent: "stellar-etl/1.0.0"})
+	if err != nil {
+		return archiveInterface, err
+	}
+
+	// Use random archive pool
+	return archivePool[int32(rand.Intn(len(archivePool)))], nil
```

Comment:
I think you should just be returning the pool itself and let it randomize on every invocation rather than picking one that’s fixed randomly.

Reply:
So just return ArchivePool.
I'm not sure how it returns a random instance though. Is that a golang thing? My guess was it was always sending the first element in the array.

Reply:
Yeah, it's a golang thing when the [NewArchivePool](https://github.com/stellar/go/blob/master/historyarchive/archive_pool.go) is created. The code actually connects to all HAS URLs and then will randomly select one of the connections when a method is called

Reply:
Yep, exactly. Basically `ArchivePool` acts **exactly** like a normal archive by providing the same interface, but under the hood it's selecting a random archive.